File: backend/CertificatesManagerApi/Mappers/CertificateTypeMapper.cs
using CertificatesManagerApi.DTOs;
using CertificatesManagerApi.Entities;

namespace CertificatesManagerApi.Mappers
{
    public class CertificateTypeMapper
    {
        public static CertificateTypeDTO ToDto(CertificateType certificateType)
        {
            CertificateTypeDTO certificateTypeDto = new CertificateTypeDTO
            {
                Handle = certificateType.Handle,
                Name = certificateType.Type
            };

            return certificateTypeDto;
        }

        public static CertificateType ToEntity(CertificateTypeDTO certificateTypeDTO)
        {
            CertificateType certificateType = new CertificateType
            {
                Handle = certificateTypeDTO.Handle,
                Type = certificateTypeDTO.Name
            };

            return certificateType;
        }
    }
}

File: frontend/src/common/utils/mappers/certificate-type-mapper.ts
import { CertificateTypeModel } from '../../models/certificate.model';
import { CertificateTypeDTO } from '../../models/dtos/certificate-type-dto';

export class CertificateTypeMapper {
	static ToModel(certificateTypeDTO: CertificateTypeDTO): CertificateTypeModel {
		let certificateTypeModel: CertificateTypeModel = {
			id: certificateTypeDTO.handle,
			type: certificateTypeDTO.name,
		};

		return certificateTypeModel;
	}
}
